eth/protocols/snap/sync.go

```diff
@@ -705,6 +705,9 @@ func (s *Syncer) loadSyncStatus() {
 					}
 				}
 			}
+			s.lock.Lock()
+			defer s.lock.Unlock()
```

Comment:
This won't fix the issue unfortunately. These fields below are updated throughout the snap sync process locklessly. They were meant to be modified only by the syner loop and not accessed from the outside. Adding a lock everywhere will become super brittle.

Trying to figure out what the best approach would be..

Reply:
- Using atomic would mess up the struct internals as they need to be 64 byte aligned not to crach on 32 bit platforms.
- Using locks everywhere to update these fields are yuck
- Requesting the progress via the syncer via a channel would lead to arbitrary timings since the syncer might be hanging on some leveldb-compaction write.

Reply:
Copy it out onto a second one (intended for external exposure)  every once in a while, and use locks on that one?